### PR TITLE
add docs about kube-registry extensions

### PR DESCRIPTION
**Pull request by freehan**

### CODE REVIEW

**Comment by thockin** [comment]
Travis says: Generated docs are out of date. Please run hack/update-generated-docs.sh

you can re-apply LGTM when you re-push

**Comment by freehan** [comment]
Extended this PR a bit. Sending all docs in one PR. @thockin please take a look.

File: cluster/addons/registry/auth/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        - name: REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY
+          value: /var/lib/registry
+        - name: REGISTRY_AUTH_HTPASSWD_REALM
+          value: basic_realm
+        - name: REGISTRY_AUTH_HTPASSWD_PATH
+          value: /auth/htpasswd
+        volumeMounts:
+        - name: image-store
+          mountPath: /var/lib/registry
+        - name: cert-dir
```
Comment (thockin) [inline cluster/addons/registry/auth/README.md]: I wouldn't call this "cert", which implies certificates

**Comment by thockin** [comment]
These are good docs - we should point people at them now that they explain how to actually DO things.

I only had a small nit, then you can self-lgtm

**Comment by freehan** [comment]
@k8s-bot unit test this please

**Pull request merged**